src/io/free_functions.rs

```diff
@@ -0,0 +1,293 @@
+use std::ffi::OsString;
+use std::fs::File;
+use std::io;
+use std::io::{BufRead, BufReader, BufWriter, Seek};
+use std::path::Path;
+use std::u32;
+
+#[cfg(feature = "bmp")]
+use bmp;
+#[cfg(feature = "gif_codec")]
+use gif;
+#[cfg(feature = "hdr")]
+use hdr;
+#[cfg(feature = "ico")]
+use ico;
+#[cfg(feature = "jpeg")]
+use jpeg;
+#[cfg(feature = "png_codec")]
+use png;
+#[cfg(feature = "pnm")]
+use pnm;
+#[cfg(feature = "tga")]
+use tga;
+#[cfg(feature = "tiff")]
+use tiff;
+#[cfg(feature = "webp")]
+use webp;
+
+use color;
+use image;
+use dynimage::DynamicImage;
+use image::{ImageDecoder, ImageFormat, ImageResult};
+use ImageError;
+
+/// Internal error type for guessing format from path.
+pub(crate) enum PathError {
+    /// The extension did not fit a supported format.
+    UnknownExtension(OsString),
+    /// Extension could not be converted to `str`.
+    NoExtension,
+}
+
+pub fn open_impl(path: &Path) -> ImageResult<DynamicImage> {
```

Comment:
This can be `pub(crate)`, right?

Reply:
`free_functions` is `pub(crate)`, hence these are public so we can choose more freely what to export under which name.

Reply:
My personal preference would be to mark only the things that are actually public as `pub` and make everything else `pub(crate)`.

Reply:
Sounds reasonable and less error prone.